armi/reactor/converters/axialExpansionChanger.py

```diff
@@ -506,12 +502,19 @@ def _getLinkedComponents(self, b, c):
 
         self.linkedComponents[c] = lstLinkedC
 
-        if lstLinkedC[0] is None:
+        if lstLinkedC[0] is None and self.linkedBlocks[b][0] is not None:
+            # only print debug if there is a linked block below in the first place
             runLog.debug(
                 f"Assembly {self.a}, Block {b}, Component {c} has nothing linked below it!",
                 single=True,
             )
-        if lstLinkedC[1] is None:
+        if (
+            lstLinkedC[1] is None
+            and self.linkedBlocks[b][1] is not None
+            and not self.linkedBlocks[b][1].hasFlags(Flags.DUMMY)
+        ):
+            # only print debug is there is a linked block above in the first place,
+            # and if that linked block is not the DUMMY block
```

Comment:
I think you might need to add specific tests for these additions. Prior coverage was provided implicitly by other tests, but since you're enforcing that these only get hit on screw ups (rightfully so) you'll likely need to add tests to provide coverage. Without them, I imagine there will be a coverage drop.